Cgen error when using unsafe on fixed array

### Describe the bug

The code:
```v
fn main(){
    x := [10]int{}
    y := unsafe {x}
    println(y)
}
```
results in the following error:
/tmp/v_1000/main.5363683437166043845.tmp.c:12262: error: '{' expected (got ";")


### Expected Behavior

Using unsafe keyword on a fixed array is not implemented but it also should not give a cgen error. @spytheman suggests it should be checker error, or a nop op

### Current Behavior

/tmp/v_1000/main.5363683437166043845.tmp.c:12262: error: '{' expected (got ";")

### Reproduction Steps

Use unsafe keyword when assigning fixed array to a new variable

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.3.3 d1f57ea

### Environment details (OS name and version, etc.)

OS: linux, "Arch Linux"
CC version: cc (GCC) 12.2.1 20230201

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -329,7 +329,8 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 		unaliased_right_sym := g.table.final_sym(unwrapped_val_type)
 		is_fixed_array_var := unaliased_right_sym.kind == .array_fixed && val !is ast.ArrayInit
 			&& (val in [ast.Ident, ast.IndexExpr, ast.CallExpr, ast.SelectorExpr]
-			|| (val is ast.CastExpr && (val as ast.CastExpr).expr !is ast.ArrayInit))
+			|| (val is ast.CastExpr && (val as ast.CastExpr).expr !is ast.ArrayInit)
+			|| (val is ast.UnsafeExpr && (val as ast.UnsafeExpr).expr is ast.Ident))
 			&& !g.pref.translated
 		g.is_assign_lhs = true
 		g.assign_op = node.op

--- a/vlib/v/tests/unsafe_fixed_array_test.v
+++ b/vlib/v/tests/unsafe_fixed_array_test.v
@@ -1,0 +1,11 @@
+fn test_fixed_array_on_unsafe_int() {
+	x := [3]int{}
+	y := unsafe { x }
+	assert x == [0, 0, 0]!
+}
+
+fn test_fixed_array_on_unsafe_string() {
+	x := [3]string{init: ''}
+	y := unsafe { x }
+	assert x == ['', '', '']!
+}

